src/components/ContactsList/index.js

```diff
@@ -33,31 +56,53 @@ export const withContacts = BaseComponent =>
     };
 
     async componentDidMount() {
-      const contactsRaw = await cozy.client
-        .fetchJSON("GET", "/data/io.cozy.contacts/_all_docs?include_docs=true")
-        .then(data => data.rows)
-        .catch(error => {
-          this.setState(state => ({ ...state, contacts: [], error }));
-        });
-      const contacts = (contactsRaw || [])
-        .map(row => ({
-          ...row.doc
-        }))
-        .filter(contact => contact.email);
+      const contacts = await cozyHTTP.getContacts();
       this.setState(state => ({
         ...state,
         contacts: [...state.contacts, ...contacts],
         loading: false
       }));
     }
 
+    deleteContact = async contact => {
+      const index = this.state.contacts.indexOf(contact);
+      this.setState(state => ({
+        ...state,
+        contacts: [
+          ...state.contacts.slice(0, index),
+          ...state.contacts.slice(index + 1)
+        ]
+      }));
+      try {
+        const result = await cozyHTTP.deleteContact(contact);
+        if (!result.deleted)
+          throw new Error("Contact not deleted on the stack");
+      } catch (ex) {
+        this.setState(state => ({
+          ...state,
+          contacts: [
+            ...state.contacts.slice(0, index),
+            contact,
+            ...state.contacts.slice(index + 1)
+          ]
+        }));
+        throw ex;
+      }
+    };
+
     render() {
       if (this.state.error) {
         return <ContactsError error={this.state.error} />;
       }
       if (this.state.loading) {
         return <div>Loading...</div>;
       }
-      return <BaseComponent contacts={this.state.contacts} {...this.props} />;
+      return (
+        <BaseComponent
+          contacts={this.state.contacts}
+          deleteContact={this.deleteContact}
+          {...this.props}
+        />
+      );
```

Comment:
The HOC is responsible to add server actions to the given component (here and now, `<ContactsApp />`, before `<ContactsList />` but now we need to share contacts between `<ContactsList />` and `<ContactCard />`).